core/dbt/events/serialization.py

```diff
@@ -0,0 +1,49 @@
+from mashumaro import DataClassDictMixin
+from mashumaro.config import (
+    BaseConfig as MashBaseConfig
+)
+from mashumaro.types import SerializationStrategy
+from datetime import datetime
+from dateutil.parser import parse
+from typing import cast
+
+
+class ExceptionSerialization(SerializationStrategy):
+    def serialize(self, value):
+        out = str(value)
+        return out
+
+    def deserialize(self, value):
+        return (Exception(value))
+
+
+class DateTimeSerialization(SerializationStrategy):
+    def serialize(self, value):
+        out = value.isoformat()
+        # Assume UTC if timezone is missing
+        if value.tzinfo is None:
+            out += "Z"
```

Comment:
let's centralize this logic re: my comment on the `Event` base class

Reply:
This is not really related to the ts calls. I'm not sure what you mean by this comment.

Reply:
Right now this PR has two places where we serialize datetimes, and they are not being serialized the same way. One here, and one in `functions.py::get_ts_rfc3339`. What I think should happen is to replace the body of this function with the contents of  `get_ts_rfc3339`, remove `get_ts_rfc3339`, and replace existing calls to it with calls to this serialization method.

Reply:
The serialization of datetimes in the events matches the serialization of datetimes that we do in the artifacts. The serialization that's done in the functions.py code is for visual presentation in the logs. These are different purposes, in my opinion.

Reply:
On second thought, I think we could use some standardization in serializing the datetimes, but I'd like to do that as a separate ticket because it requires moving around some things that are not related to this pull request. I've opened #4608 for that.